src/test/java/org/jenkinsci/plugins/docker/workflow/DockerDSLTest.java

```diff
@@ -423,4 +423,54 @@ private static void grep(File dir, String text, String prefix, Set<String> match
             }
         });
     }
+
+    @Issue("JENKINS-26178")
+    @Test public void container_exec() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                assumeDocker();
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "build");
+                p.setDefinition(new CpsFlowDefinition(
+                    "node {\n" +
+                    "  writeFile file: 'Dockerfile', text: '# This is a test.\\n\\nFROM alpine\\nRUN chmod u+s /bin/busybox \\nENTRYPOINT [\\\"/bin/ping\\\" ] \\nCMD [\\\"localhost\\\"] \\n'\n" +
+                    "  def built = docker.build 'alpine-entrypoint-test'\n" +
+                    "  echo \"built ${built.id}\"\n" +
+                    "}", true));
+                WorkflowRun b = story.j.assertBuildStatusSuccess(p.scheduleBuild2(0));
+            }
+        });
+        story.addStep(new Statement() {
```

Comment:
Any particular reason you are doing two Jenkins restarts in the middle of this test? It seems gratuitous, since you are not testing resumption of builds.

Just fold these all into one step.